packages/astro/src/env/env-loader.ts

```diff
@@ -30,7 +31,7 @@ function getPrivateEnv(
 				}
 				// Boolean values should be inlined to support `export const prerender`
 				// We already know that these are NOT sensitive values, so inlining is safe
-				if (value === '0' || value === '1' || value === 'true' || value === 'false') {
+				if (coerceValues && (value === '0' || value === '1' || value === 'true' || value === 'false')) {
```

Comment:
FYI this code existed before `astro:env` so I'm afraid this change would be a breaking one. It would be great if you could test on an older version of Astro. To be sure, in `4.9`since `astro:env` was introduced in `4.10`

Reply:
I'm a bit confused sorry. Is it a breaking change because people may be relying on these values being converted? And what do you mean by testing on an older version of Astro?

Reply:
Sorry I guess I was not clear enough. I indeed think it would be a breaking change if people relied on the coercion. In this case that would be a change for Astro 6 later this year.

> And what do you mean by testing on an older version of Astro?

Basically I'm wondering if this coercion code has always been there or not (ie. before `astro:env` was introduced). So by checking an older version of Astro (before the introduction of `astro:env`), we can check if the coercion was already done. If not, we'll probably have to check a few versions like in 5.0 or something to see since when this coercion has been there.

Let me know if I can clarify anything!

Reply:
It looks like the coercion logic was added in a9373c0 but the `vite-plugin-import-meta-env` module didn't exist until 80067c0, which makes me think that the coercion on `import.meta.env` started happening then.

This commit was released with [astro@5.1.6](https://github.com/withastro/astro/releases/tag/astro%405.1.6).

Reply:
Sorry for the comment spam

Did a little more digging. The coercion logic was added in #6910 which was first included in [astro@2.3.4](https://github.com/withastro/astro/releases/tag/astro@2.3.4).

It looks like this was implemented to allow people to use an environment variable with `export const prerender` (see #6772).

I feel like that's the wrong approach though. Whatever is reading the `prerender` export should be the thing coercing the value.

Reply:
Thanks for the investigation! Yeah, [support for dynamic prerender was removed in v5](https://docs.astro.build/en/guides/upgrade-to/v5/#removed-support-for-dynamic-prerender-values-in-routes). 

Given since when the coercion exists, I think that will require a major anyways. Let's see if we can do it in v6: https://github.com/withastro/roadmap/discussions/1174#discussioncomment-13543135